src/scales/scale.category.js

```diff
@@ -49,10 +50,19 @@ module.exports = Scale.extend({
 
 	getLabelForIndex: function(index, datasetIndex) {
 		var me = this;
-		var data = me.chart.data;
+		var chart = me.chart;
+		var data = chart.data;
 		var isHorizontal = me.isHorizontal();
+		var controller = chart.getDatasetMeta(datasetIndex).controller;
 
-		if (data.yLabels && !isHorizontal) {
+		// For controllers supporting getValueScaleId, we can be sure if this is a value scale.
+		// For object data ({x,y} etc.) we can rely on getRightValue to return correct thing.
+		// For others, check if data is a label and use that instead (so this is an value index, but values are labels)
+		var isValueScale = controller.getValueScaleId
```

Comment:
Does this variable name need to be updated? It doesn't seem like we're just checking that this scale is the controller's value scale anymore